WooCommerce/Classes/ViewRelated/Products/Edit Product/Downloads/Downloadable FIle List/ProductDownloadListViewController+Droppable.swift

```diff
@@ -0,0 +1,50 @@
+import UIKit
+import CoreServices
+import Yosemite
+
+// MARK: - UITableViewDropDelegate
+//
+extension ProductDownloadListViewController: UITableViewDropDelegate {
+    func tableView(_ tableView: UITableView, canHandle session: UIDropSession) -> Bool {
+        return session.canLoadObjects(ofClass: ProductDownloadDnD.self) || session.hasItemsConforming(toTypeIdentifiers: [kUTTypeUTF8PlainText as String])
+    }
+
+    func tableView(_ tableView: UITableView,
+                   dropSessionDidUpdate session: UIDropSession,
+                   withDestinationIndexPath destinationIndexPath: IndexPath?) -> UITableViewDropProposal {
+        let operation: UIDropOperation
+        if session.items.count > 1 {
+            operation = .cancel
+        } else {
+            operation = .move
+        }
+        return UITableViewDropProposal(operation: operation, intent: .insertAtDestinationIndexPath)
+    }
+
+    func tableView(_ tableView: UITableView, performDropWith coordinator: UITableViewDropCoordinator) {
+        let destinationIndexPath: IndexPath
+        if let indexPath = coordinator.destinationIndexPath {
+            destinationIndexPath = indexPath
+        } else {
+            // Get last index path of table view.
+            let section = tableView.numberOfSections - 1
+            let row = tableView.numberOfRows(inSection: section)
```

Comment:
In reality, it's 💯 safe. Because you need at least 2 items and 1 section to be able to Drag and Drop. If we have 0 or 1 file then that part of the code won't get executed at all. I have tested it well. But it's safe to add an extra check.